### ISSUE TITLE
snapcraft build for armhf

### ISSUE THREAD

**Issue by jonathantneal** [cross-platform/apple-music-for-linux#21]
Would you be willing to include a build for armhf so that we can run this snap from Raspberry Pi devices?

**Comment by MarcusTomlinson** [comment]
Hey @jonathantneal, would you mind giving this build a try on your RPi and let me know how it is:

`snap install apple-music-for-linux --candidate`

**Comment by jonathantneal** [comment]
Yes, I will do it this evening, @MarcusTomlinson, and I’ll let you know how it goes. Thanks!

**Comment by jonathantneal** [comment]
Thank you!
The installer works perfectly.
The program launches perfectly.
The program requests that I install Widevine, and that installation works perfectly.

The player itself does not work — unfortunately alerting the “_Failed to load because no supported source was found_” message. However, I believe this is a problem related to Chromium and Widevine. The same issue occurs in the Chromium that ships.

If I find a solution you can integrate, tho, I can let you know in a separate issue. For what it’s worth, **Chromium (Media Edition)** does work. I’m not really sure what that project is doing differently, but Apple Music will play through that version of the browser.

So, if/once the issue related to Chromium is resolved, your release _will immediately work_. 🎉 Thank you, again!

Feel free to direct me to try more things, or to close this if you are comfortable with this level of resolution (which, I am).

**Comment by MarcusTomlinson** [comment]
Ok so the issue here is that there's no easy way to download a legitimate arm build of Widevine. The only way I can see doing this is to download and mount a ChromeOS recovery image (roughly 1GB) and extract the required files, but that'd be a terrible user experience. I'll have to think about this more (probably not soon though, sorry).

In the meantime, you could try obtaining a Widevine arm build yourself and placing in the `~/snap/apple-music-for-linux/current/.local/lib/WidevineCdm/_platform_specific/linux_arm` to see if that works. You may also need an updated `~/snap/apple-music-for-linux/current/.local/lib/WidevineCdm/manifest.json`.

**Comment by Actionparsnip** [comment]
I have the "Failed to load because no supported source was found” error. I used the Widevine installer as suggested by the app. I can play albums as expected but i cannot play playlists that I have made of songs available on the service. Can you advise any please?

**Comment by MarcusTomlinson** [comment]
@Actionparsnip armhf does not work correctly yet unfortunately. See my previous comment.

**Comment by Actionparsnip** [comment]
> @Actionparsnip armhf does not work correctly yet unfortunately. See my previous comment.

Thanks. No worries. Is it OK on arm64 or is that the same story?

**Comment by MarcusTomlinson** [comment]
> Is it OK on arm64 or is that the same story?

Same story. Sorry.

**Comment by Actionparsnip** [comment]
Thanks. I really appreciate the replies :)

**Comment by sorriso93** [comment]
Following

**Comment by mavonarx** [comment]
Asahi Linux provides a widevine installer for arm
